libarchive/archive_read_support_format_rar.c

```diff
@@ -3681,7 +3681,7 @@ execute_filter_e8(struct rar_filter *filter, struct rar_virtual_machine *vm, siz
     {
       uint32_t currpos = (uint32_t)pos + i + 1;
       int32_t address = (int32_t)vm_read_32(vm, i + 1);
-      if (address < 0 && currpos >= -(uint32_t)address)
+      if (address < 0 && currpos >= (uint32_t)(-address))
```

Comment:
This seems like it would change the meaning - it cannot result in a negative number whereas it *could* (?) before

Reply:
That's the warning - negating an unsigned value produces a non-negative unsigned number

Reply:
I think we may have conflicting compiler warnings.   This was just changed the other way in #2235 a few days ago.  CC: @nekopsykose

Reply:
I wonder if the right answer here is to unroll the negation operation to
```
    1 + ~(uint32_t)address
```
Which is bitwise identical to twos-complement negation but should not confuse any compiler's understanding of signed vs. unsigned operations.

Reply:
changing it back to putting the `-` next to address will just make it crash in ubsan again like before

what compilation warning do you get? `-(unsigned)` is valid and afaict was the original intent (and works fine). i guess using `~` could be more explicit..

Reply:
>I wonder if the right answer here is to unroll the negation operation to 1 + ~(uint32_t)address

yeah, seems like that should make everything work too

Reply:
The warning is:
```
warning C4146: unary minus operator applied to unsigned type, result still unsigned
```
See: https://godbolt.org/z/bbv43aWjr

I can switch to using unary negation tomorrow